File: back-end/src/main/java/com/ocire/lms/service/LearningService.java
package com.ocire.lms.service;

import java.util.List;

import com.ocire.lms.dto.InsertResDto;
import com.ocire.lms.dto.learning.LearningInsertReqDto;
import com.ocire.lms.dto.learning.LearningResDto;

public interface LearningService {
	List<LearningResDto> getLearningByClassId(Long classId);
	
	InsertResDto insertLearning(LearningInsertReqDto data);
}

File: front-end/src/app/services/learning.service.ts
import { Injectable } from '@angular/core';
import { BaseModule } from '../components/base/base.module';
import { LearningInsertReqDto } from '@dto/learning/learning-insert.req.dto';
import { Observable } from 'rxjs';
import { InsertResDto } from '@dto/InsertResDto';
import { BaseService } from './base.service';
import { BASE_URL } from '../constant/api.constant';
import { LearningResDto } from '@dto/learning/learning.res.dto';

@Injectable({
  providedIn: 'root',
})
export class LearningService {
  constructor(private base: BaseService) {}

  insert(data: LearningInsertReqDto): Observable<InsertResDto> {
    return this.base.post(`${BASE_URL}/learning`, data);
  }

  getByClassId(id: number | null): Observable<LearningResDto[]> {
    return this.base.get<LearningResDto[]>(`${BASE_URL}/learning?id=${id}`);
  }
}
